src/core/net/netif.cpp

```diff
@@ -325,6 +327,97 @@ ThreadError Netif::RemoveUnicastAddress(const NetifUnicastAddress &aAddress)
     return error;
 }
 
+
+ThreadError Netif::AddExternalUnicastAddress(const NetifUnicastAddress &aAddress)
+{
+    ThreadError error = kThreadError_None;
+    int8_t index = 0;
+
+    for (NetifUnicastAddress *cur = mUnicastAddresses; cur; cur = cur->GetNext())
+    {
+        if (memcmp(&cur->mAddress, &aAddress.mAddress, sizeof(otIp6Address)) == 0)
+        {
+            VerifyOrExit(GetExtUnicastAddressIndex(cur) != -1, error = kThreadError_InvalidArgs);
+
+            cur->mPreferredLifetime = aAddress.mPreferredLifetime;
+            cur->mValidLifetime = aAddress.mValidLifetime;
+            cur->mPrefixLength = aAddress.mPrefixLength;
+            ExitNow(error = kThreadError_Busy);
+        }
+    }
+
+    // Make sure we haven't reached our limit already
+    VerifyOrExit(mCountExtUnicastAddresses != OPENTHREAD_CONFIG_MAX_EXT_IP_ADDRS, error = kThreadError_NoBufs);
```

Comment:
Can we just check to see if `mMaskExtUnicastAddresses` bit-mask is all 1's, rather than keeping a separate `mCountExtUnicastAddresses`?